### ISSUE TITLE
DnD from a modal dialog interferes with global focus [X11]

### ISSUE THREAD

**Issue by flopp** [petasis/tkdnd#40]
If a DnD operation is started from a modal dialog (implemented using `grab`/`ungrab`), I cannot get keyboard focus in other applications, until the modal dialog is closed (or some other events occur in the modal dialog).

Example program:

```
#!/usr/bin/env wish

set dir tkdnd/library
source $dir/tkdnd.tcl
::tkdnd::initialise $dir libtkdnd2.9.2.so tkdnd

proc drag_init {} {
    return {copy DND_Text some_data}
}

proc modal_dialog {} {
    set w .modal_dialog
    toplevel $w
    wm title $w "Modal Dialog"
    set text \
        "DnD Source in Modal Dialog:\n\
         (1) Press mouse button here\n\
         (2) Move the mouse around until DnD cursor occurs\n\
         (3) Release the mouse\n\
         ==> Unable to get keyboard focus in other applications until\n\
         (a) the modal dialog is closed, or\n\
         (b) this label is clicked."
    ttk::label $w.source -text $text
    pack $w.source -expand 1 -fill both

    tkdnd::drag_source register $w.source
    bind $w.source <<DragInitCmd>> {drag_init}
 
    wm transient $w .
    raise $w
    focus $w
    grab  $w
    set sentinel {}
    wm protocol $w WM_DELETE_WINDOW [list set sentinel "closed"]
    tkwait variable sentinel
    grab release $w
    destroy $w
}

ttk::button .button -text "Open Modal Dialog" -command "modal_dialog"
pack .button -expand 1 -fill both
```

**Comment by flopp** [comment]
Always returning "0" from `xdnd::_process_drag_events` fixes the issue for me...

**Comment by petasis** [comment]
Bug fixed. Thanks for the detailed example.
